api/types/types.proto

```diff
@@ -3625,4 +3624,13 @@ message ConnectionDiagnosticV1 {
 message ConnectionDiagnosticSpecV1 {
     bool Success = 1 [ (gogoproto.jsontag) = "success" ];
     string Message = 2 [ (gogoproto.jsontag) = "message" ];
+    repeated ConnectionDiagnosticTrace Traces = 3 [ (gogoproto.jsontag) = "traces" ];
+}
+
+// ConnectionDiagnosticTrace describes a trace of a connection diagnostic
+message ConnectionDiagnosticTrace {
+    string ID = 1 [ (gogoproto.jsontag) = "id" ];
+    string TraceType = 2 [ (gogoproto.jsontag) = "trace_type" ];
+    string Status = 3 [ (gogoproto.jsontag) = "status" ];
+    string Details = 4 [ (gogoproto.jsontag) = "details" ];
```

Comment:
How about adding an `Error` field that will contain an actual error message like we discussed before?

Reply:
Added :+1: